src/base-traits.jl

```diff
@@ -1,15 +1,17 @@
 module BaseTraits
 using SimpleTraits
 
-export IsLeafType, IsBits, IsImmutable, IsContiguous, IsFastLinearIndex,
+using Compat
+
+export IsLeafType, IsBits, IsImmutable, IsContiguous, IsIndexLinear,
        IsAnything, IsNothing, IsCallable
 
 "Trait which contains all types"
 @traitdef IsAnything{X}
 @traitimpl IsAnything{X} <- (x->true)(X)
 
 "Trait which contains no types"
-typealias IsNothing{X} Not{IsAnything{X}}
+@compat IsNothing{X} = Not{IsAnything{X}}
```

Comment:
It doesn't appear to be necessary (see Julia's NEWS.md file), but it doesn't seem to hurt either. Most but not all of them in https://github.com/JuliaLang/julia/pull/20500/files seem to use `const`, so I can add it if you prefer.